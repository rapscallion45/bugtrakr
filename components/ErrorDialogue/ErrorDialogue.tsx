import { FC } from 'react';
import { styled } from '@mui/styles';
import Box from '@mui/material/Box';
import WarningIcon from '@mui/icons-material/Warning';
import Typography from '@mui/material/Typography';

const BoxWrapper = styled(Box)({
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
});

const BoxInner = styled(Box)({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  padding: '100px 20px',
  maxWidth: '350px',
});

interface ErrorDialogueProps {
  displayText?: string;
}

const ErrorDialogue: FC<ErrorDialogueProps> = function ErrorDialogue({ displayText = null }) {
  return (
    <BoxWrapper>
      <BoxInner>
        <WarningIcon fontSize="large" color="error" sx={{ margin: '15px 0' }} />
        {displayText && (
          <Typography variant="h6" component="h6" align="center">
            {displayText}
          </Typography>
        )}
      </BoxInner>
    </BoxWrapper>
  );
};

export default ErrorDialogue;
