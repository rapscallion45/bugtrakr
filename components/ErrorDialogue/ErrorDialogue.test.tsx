import React from 'react';
import { mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import ErrorDialogue from './ErrorDialogue';

const testText = 'This is the test error message.';

describe('Error Dialogue', () => {
  it('Renders correctly enzyme', () => {
    /* Arrange */
    /* Act */
    const view = render(<ErrorDialogue displayText={testText} />);

    /* Assert */
    expect(toJson(view)).toMatchSnapshot();
  });

  describe('Input Text', () => {
    it('Should render the passed input text', () => {
      /* Arrange */
      /* Act */
      const view = mount(<ErrorDialogue displayText={testText} />);

      /* Assert */
      expect(view.find('h6').text()).toEqual(testText);
    });
  });
});
