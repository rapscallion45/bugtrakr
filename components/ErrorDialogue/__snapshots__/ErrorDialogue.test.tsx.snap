// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Dialogue Renders correctly enzyme 1`] = `
<div
  class="ForwardRef(Box)-root-1 MuiBox-root css-0"
>
  <div
    class="ForwardRef(Box)-root-2 MuiBox-root css-0"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeLarge css-1dp2k9f-MuiSvgIcon-root"
      data-testid="WarningIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
      />
    </svg>
    <h6
      class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-1kgxqm0-MuiTypography-root"
    >
      This is the test error message.
    </h6>
  </div>
</div>
`;
